### ISSUE TITLE
Allow feeding in frequency-dependent gain factor for broadband calibration

### ISSUE THREAD

**Issue by leewujung** [OSOceanAcoustics/echopype#458]
While working on #446 it became obvious that there is a need for users to be able to put in frequency-dependent gain factors externally and not relying on what is or isn't in the data. This is especially true since a lot of files don't already have the frequency-dependent gain factors stored (i.e. the `Vendor` group does not have a data variable called `gain` with dimension `cal_channel_id` and `cal_frequency`).

This seems can be handled in 2 ways:
1. Similar to how we update the platform group to add missing GPS data using `echodata.update_platform()`
2. Feeding in a `gain` xarray DataArray as part of the input argument `cal_params` when calling `compute_Sv`

Pinging @emiliom here since the route we choose is related to what we consider should be an integral part of the data file. 
Perhaps both should be supported? Users may want to apply different gain factors from different calibration sessions for comparison, so route 2 is useful.

**Comment by emiliom** [comment]
Are there specific, convention defined variables for gain factors? Can you point me to them (name(s) and group)

**Comment by leewujung** [comment]
It's a little involved...

There is a `gain_correction` data variable under the `Beam` group, which I decided to use to store the gain factor for CW mode data. There is another `gain_correction` data variable in the `Vendor` group that I decided to store, because it contains info that is more comprehensive than that stored under the `Beam` group (basically this is a look up table with coordinate dimension `frequency` and `pulse_length` -- the `gain_correction` in the `Beam` group is the result of indexing using the actual `pulse_length`)

For broadband data there could be additional calibration results (frequency-dependent gain factors) saved in the config XML, which are currently stored under data variable `gain` in the `Vendor` group with coordinate dimensions `cal_channel_id` and `cal_frequency`. The data variable name `gain` here should probably be changed to something like `gain_correction_broadband` or something similar to be consistent but distinguishable from the `gain_correction` defined in the convention.

Note for BB data files the narrowband gain (the ones used for CW data) still exist, probably because the echosounder can function in both modes so it is more convenient for them to just be included regardless.
